### PR TITLE
Wip update24 rk

### PR DESCRIPTION
**Pull request by rkuhn**

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @paulcleary,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://www.typesafe.com/contribute/cla](http://www.typesafe.com/contribute/cla)

File: src/main/scala/akka/persistence/dynamodb/journal/DynamoDBRequests.scala [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+   * a Future.success(Failure) as the result.  This is needed to be compliant with
+   * Akka Persistence 2.4
+   *
+   * @param atomicWrite Contains a list of persistentRepr that need to be persisted atomically
+   * @return [[Future[Try[Unit]] - This is important, we always must return a result, even on failure
+   */
+  def writeMessages(atomicWrite: AtomicWrite): Future[Try[Unit]] = {
+
+    // groups messages into groups of 12, a BatchWriteItemRequest has a max of 25 items, and
+    // each PersistentRepr has to writes, the most we can put in any one request is 12 PersistentRepr
```
Comment (paulcleary) [inline src/main/scala/akka/persistence/dynamodb/journal/DynamoDBRequests.scala]: comment - "has two writes", that seems like my bad

Comment (rkuhn) [inline src/main/scala/akka/persistence/dynamodb/journal/DynamoDBRequests.scala]: Hmm, why isn’t this shown as a rename? Anyway, I’ll fix it. (my plan was to go through once everything is updated and working again—got some failures with 2.4.1 and AWS SDK 1.20.50 on Travis)

**Comment by paulcleary** [comment]
One additional thought on this PR, would be nice if the table was automatically created for us when the system started up (I saw other plugins to the same).

If we auto-create the table, it would make sense to define the throughput for the table in the config file

**Comment by rkuhn** [comment]
Good point, filed #18

**Comment by lightbend-cla-validator** [comment]
Hi @paulcleary,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Typesafe Contributors License Agreement:

[http://www.typesafe.com/contribute/cla](http://www.typesafe.com/contribute/cla)

**Pull request merged**